---
title: "Computational Predictor Processing"
author: "Garrett Cooper"
date: 12/02/2024
output:
  github_document:
    toc: true
    html_preview: false
editor_options: 
  chunk_output_type: inline
---

```{r}
library(yaml)
library(stringr)
library(dplyr)

#read in config file
config <- read_yaml("./config.yaml")

```


Lets first filter the CADD v1.7 variants


```{r}

library(readr)
library(dplyr)


cadd_data <- read_tsv(file=config$cadd_raw)

cadd_data_exons <- cadd_data %>%
  filter(Pos >= 23787170 & Pos <= 23787262 | #exon 1
         Pos >= 23791756 & Pos <= 23791894 | #exon 2
         Pos >= 23793559 & Pos <= 23793688 | #exon 3
         Pos >= 23800944 & Pos <= 23801081 | #exon 4
         Pos >= 23803295 & Pos <= 23803422 | #exon 5
         Pos >= 23816770 & Pos <= 23816936 | #exon 6
         Pos >= 23825225 & Pos <= 23825415 | #exon 7
         Pos >= 23833572 & Pos <= 23833703 | #exon 8
         Pos >= 23834141 & Pos <= 23834180)  #exon 9




cadd_data_exons



```


```{r}


# Add a column that increases by 1 every 3 rows starting at 1
cadd_data_exons$Site <- rep(1:ceiling(nrow(cadd_data_exons)/3), each = 3, length.out = nrow(cadd_data_exons))


cadd_data_exons <- cadd_data_exons %>%
  # Create a new grouping column (block of 3 sites: 1-3, 4-6, etc.)
  mutate(Codon = (Site - 1) %/% 3 + 1) %>%
  group_by(Codon) %>%
  # Concatenate unique Ref values per group
  mutate(Ref_Codon = paste(Ref, collapse = "")) %>%
  ungroup() %>%
  mutate(Ref_Codon = str_sub(Ref_Codon, 2, 2) %>% 
           paste0(str_sub(Ref_Codon, 4, 4), 
                  str_sub(Ref_Codon, 7, 7)))

cadd_data_exons <- cadd_data_exons %>%
  mutate(codon_site = (Site - 1) %% 3 + 1,  # Create a new column with the adjusted Site values
         Alt_Codon = mapply(function(codon, site, alt) {
           # Convert codon into a vector of characters
           codon_vec <- strsplit(codon, "")[[1]]
           
           # Replace the base at the position specified by 'site'
           codon_vec[site] <- alt
           
           # Recombine the codon and return it
           paste0(codon_vec, collapse = "")
         }, Ref_Codon, codon_site, Alt))  # Use the new 'codon_site' column


codon_table <- c(
  "ATA" = "I", "ATC" = "I", "ATT" = "I", "ATG" = "M", "ACA" = "T", "ACC" = "T",
  "ACG" = "T", "ACT" = "T", "AAC" = "N", "AAT" = "N", "AAA" = "K", "AAG" = "K",
  "AGC" = "S", "AGT" = "S", "AGA" = "R", "AGG" = "R", "CTA" = "L", "CTC" = "L",
  "CTG" = "L", "CTT" = "L", "CCA" = "P", "CCC" = "P", "CCG" = "P", "CCT" = "P",
  "CAC" = "H", "CAT" = "H", "CAA" = "Q", "CAG" = "Q", "CGA" = "R", "CGC" = "R",
  "CGG" = "R", "CGT" = "R", "GTA" = "V", "GTC" = "V", "GTG" = "V", "GTT" = "V",
  "GCA" = "A", "GCC" = "A", "GCG" = "A", "GCT" = "A", "GAC" = "D", "GAT" = "D",
  "GAA" = "E", "GAG" = "E", "GGA" = "G", "GGC" = "G", "GGG" = "G", "GGT" = "G",
  "TCA" = "S", "TCC" = "S", "TCG" = "S", "TCT" = "S", "TTC" = "F", "TTT" = "F",
  "TTA" = "L", "TTG" = "L", "TAC" = "Y", "TAT" = "Y", "TAA" = "Stop", "TAG" = "Stop",
  "TGC" = "C", "TGT" = "C", "TGA" = "Stop", "TGG" = "W"
)

# Create a new column to report the amino acid substitution
cadd_data_exons <- cadd_data_exons %>%
  mutate(Amino_Acid_Ref = sapply(Ref_Codon, function(codon) codon_table[codon]),
         Amino_Acid_Alt = sapply(Alt_Codon, function(codon) codon_table[codon]),
         AA_Substitution = paste(Amino_Acid_Ref, Codon, Amino_Acid_Alt, sep=""))


cadd_data_exons <- cadd_data_exons %>%
  mutate(Mutation_type = case_when(
    # Stop codons (excluding stop codons replaced with an amino acid)
    grepl("Stop", AA_Substitution) & !grepl("^X\\d+[A-Za-z]$", AA_Substitution) & !grepl("Stop\\d+[A-Za-z]$", AA_Substitution) ~ "Stop", 
    # Special case: Stop codon replaced with an amino acid (e.g., X386Q, Stop386Q)
    grepl("^X\\d+[A-Za-z]$", AA_Substitution) | grepl("Stop\\d+[A-Za-z]$", AA_Substitution) ~ "M",  
    # Synonymous mutation (where first and last characters are the same)
    substr(AA_Substitution, 1, 1) == substr(AA_Substitution, nchar(AA_Substitution), nchar(AA_Substitution)) ~ "S", 
    # All others (non-synonymous)
    TRUE ~ "M"
  ))

#We also need to reformat the CADD data to match with the mitedata
cadd_data_exons <- cadd_data_exons %>%
  mutate(AA_Substitution = str_replace(AA_Substitution, "Stop", "X"))


write.table(cadd_data_exons, file = "./data/computational_predictors/SMARCB1_CADD_v1.7_annotated.tsv", sep = "\t", row.names = FALSE, quote = FALSE)

#CADD has duplicate rows where multiple nucleotide substitution lead to the same amino acid change. Here we will average the PHRED score across all duplicates and keep as one row.
cadd_data_no_duplicates <- cadd_data_exons %>%
  group_by(AA_Substitution) %>%
  summarise(
    across(-PHRED, ~ first(.), .names = "{.col}"), # Keep the first instance for all columns except PHRED
    PHRED = mean(PHRED, na.rm = TRUE) # Calculate the average for PHRED
  ) %>%
  ungroup()


#Write this to a tsv for plotting later in the heatmap and use for ROC curve predictions

cadd_data_no_duplicates[2682, "Mutation_type"] <- "S"

write_tsv(cadd_data_no_duplicates, "./data/computational_predictors/duplicates_averaged/SMARCB1_CADD_v1.7_duplicates_averaged.tsv")



#Here we are going to average all mutations at each residue to get a metric of general mutational intolerance for each residue

cadd_data_exons_missense <- cadd_data_no_duplicates %>%
  filter(cadd_data_no_duplicates$Mutation_type == "M")

write_tsv(cadd_data_exons_missense, "./data/computational_predictors/SMARCB1_CADD_v1.7_missense_only.tsv")

cadd_pathogenic <- cadd_data_exons_missense %>%
  filter(cadd_data_exons_missense$PHRED > 20)

cadd_ambiguous <- cadd_data_exons_missense %>%
  filter(cadd_data_exons_missense$PHRED < 20 & cadd_data_exons_missense$PHRED > 10)

cadd_benign <- cadd_data_exons_missense %>%
  filter(cadd_data_exons_missense$PHRED < 10)
 

cadd_average_residue <- cadd_data_exons_missense %>%
  group_by(Codon) %>%
  summarise(res_avg_PHRED = mean(PHRED, na.rm = TRUE))



cadd_average_residue_sorted <- cadd_average_residue[order(-cadd_average_residue$res_avg_PHRED), ]

cadd_average_residue_sorted$rank <- 1:nrow(cadd_average_residue_sorted)

write_tsv(cadd_average_residue_sorted, "./data/computational_predictors/residue_averaged_zscores/CADD_SMARCB1_v1.7_res_avg.tsv")




```




Next lets filter the REVEL v1.3 Variants


```{r}

library(dplyr)

revel_data <- read.csv(file=config$revel_raw)


revel_data_filtered <- revel_data %>%
  filter(grch38_pos >= 23787169 & grch38_pos <= 23834181)


revel_data_filtered <- revel_data_filtered[,-2]

#Reformat to match CADD data
colnames(revel_data_filtered)[colnames(revel_data_filtered) == "grch38_pos"] <- "Pos"

revel_data_filtered$Pos <- as.numeric(revel_data_filtered$Pos)

```

```{r}

revel_data_exons <- revel_data_filtered %>%
  filter(Pos >= 23787170 & Pos <= 23787262 | #exon 1
         Pos >= 23791756 & Pos <= 23791894 | #exon 2
         Pos >= 23793559 & Pos <= 23793688 | #exon 3
         Pos >= 23800944 & Pos <= 23801081 | #exon 4
         Pos >= 23803295 & Pos <= 23803422 | #exon 5
         Pos >= 23816770 & Pos <= 23816936 | #exon 6
         Pos >= 23825225 & Pos <= 23825415 | #exon 7
         Pos >= 23833572 & Pos <= 23833703 | #exon 8
         Pos >= 23834141 & Pos <= 23834180)  #exon 9

```


```{r}

#Since revel does not have every site like cadd, we are going to use cadd as a reference to pull in some data
cadd_data_exons_unique <- cadd_data_exons %>%
  distinct(Pos, .keep_all = TRUE)

revel_data_exons <- revel_data_exons %>%
  left_join(cadd_data_exons_unique %>% select(Pos, Codon, Site), by = "Pos")



cadd_data_exons_unique_codon <- cadd_data_exons %>%
  distinct(Codon, .keep_all = TRUE)

cadd_data_exons_unique_codon$Codon <- as.numeric(cadd_data_exons_unique_codon$Codon)

revel_data_exons <- revel_data_exons %>%
  left_join(cadd_data_exons_unique_codon %>% select(Codon, Ref_Codon), by = "Codon")




revel_data_exons <- revel_data_exons %>%
  mutate(codon_site = (Site - 1) %% 3 + 1,  # Create a new column with the adjusted Site values
         Alt_Codon = mapply(function(codon, site, alt) {
           # Convert codon into a vector of characters
           codon_vec <- strsplit(codon, "")[[1]]
           
           # Replace the base at the position specified by 'site'
           codon_vec[site] <- alt
           
           # Recombine the codon and return it
           paste0(codon_vec, collapse = "")
         }, Ref_Codon, codon_site, alt))  # Use the new 'codon_site' column

codon_table <- c(
  "ATA" = "I", "ATC" = "I", "ATT" = "I", "ATG" = "M", "ACA" = "T", "ACC" = "T",
  "ACG" = "T", "ACT" = "T", "AAC" = "N", "AAT" = "N", "AAA" = "K", "AAG" = "K",
  "AGC" = "S", "AGT" = "S", "AGA" = "R", "AGG" = "R", "CTA" = "L", "CTC" = "L",
  "CTG" = "L", "CTT" = "L", "CCA" = "P", "CCC" = "P", "CCG" = "P", "CCT" = "P",
  "CAC" = "H", "CAT" = "H", "CAA" = "Q", "CAG" = "Q", "CGA" = "R", "CGC" = "R",
  "CGG" = "R", "CGT" = "R", "GTA" = "V", "GTC" = "V", "GTG" = "V", "GTT" = "V",
  "GCA" = "A", "GCC" = "A", "GCG" = "A", "GCT" = "A", "GAC" = "D", "GAT" = "D",
  "GAA" = "E", "GAG" = "E", "GGA" = "G", "GGC" = "G", "GGG" = "G", "GGT" = "G",
  "TCA" = "S", "TCC" = "S", "TCG" = "S", "TCT" = "S", "TTC" = "F", "TTT" = "F",
  "TTA" = "L", "TTG" = "L", "TAC" = "Y", "TAT" = "Y", "TAA" = "Stop", "TAG" = "Stop",
  "TGC" = "C", "TGT" = "C", "TGA" = "Stop", "TGG" = "W"
)


revel_data_exons <- revel_data_exons %>%
  mutate(Amino_Acid_Ref = sapply(Ref_Codon, function(codon) codon_table[codon]),
         Amino_Acid_Alt = sapply(Alt_Codon, function(codon) codon_table[codon]),
         AA_Substitution = paste(Amino_Acid_Ref, Codon, Amino_Acid_Alt, sep=""))

#One site at exon 5 was incorrectly annotated will be filtered
revel_data_exons <- revel_data_exons %>%
  filter(revel_data_exons$AA_Substitution != "C167Stop")


revel_data_exons <- revel_data_exons %>%
  mutate(Mutation_type = case_when(
    # Special case: Stop codon replaced with an amino acid (e.g., X386Q, Stop386Q)
    grepl("^X\\d+[A-Za-z]$", AA_Substitution) | grepl("Stop\\d+[A-Za-z]$", AA_Substitution) ~ "M",  
    # Synonymous mutation (where first and last characters are the same)
    substr(AA_Substitution, 1, 1) == substr(AA_Substitution, nchar(AA_Substitution), nchar(AA_Substitution)) ~ "S", 
    # All others (non-synonymous)
    TRUE ~ "M"
  ))

#There were some inaccurately annotated regions, lets just take them out.
revel_data_exons <- revel_data_exons %>%
  filter(revel_data_exons$Mutation_type == "M")




write.table(revel_data_exons, file = "./data/computational_predictors/SMARCB1_revel_v1.3_annotated.tsv", sep = "\t", row.names = FALSE, quote = FALSE)

#REVEL has duplicate rows where multiple nucleotide substitution lead to the same amino acid change. Here we will average the REVEL score across all duplicates and keep as one row.
revel_data_no_duplicates <- revel_data_exons %>%
  group_by(AA_Substitution) %>%
  summarise(
    across(-REVEL, ~ first(.), .names = "{.col}"), # Keep the first instance of all columns
    REVEL = mean(REVEL, na.rm = TRUE) # Replace REVEL with the average of duplicates
  ) %>%
  ungroup()

#Save this later so we can plot this in a heatmap and use for ROC curve predictions

write_tsv(revel_data_no_duplicates, "./data/computational_predictors/duplicates_averaged/SMARCB1_REVEL_v1.3_duplicates_averaged.tsv")


revel_pathogenic <- revel_data_no_duplicates %>%
  filter(revel_data_no_duplicates$REVEL > 0.75)

revel_ambiguous <- revel_data_no_duplicates %>%
  filter(revel_data_no_duplicates$REVEL > 0.5 & revel_data_no_duplicates$REVEL < 0.75)

revel_benign <- revel_data_no_duplicates %>%
  filter(revel_data_no_duplicates$REVEL < 0.5)

 

revel_average_residue <- revel_data_no_duplicates %>%
  group_by(Codon) %>%
  summarise(res_avg_REVEL = mean(REVEL, na.rm = TRUE))



revel_average_residue_sorted <- revel_average_residue[order(-revel_average_residue$res_avg_REVEL), ]

revel_average_residue_sorted$rank <- 1:nrow(revel_average_residue_sorted)

write_tsv(revel_average_residue_sorted, "./data/computational_predictors/residue_averaged_zscores/REVEL_SMARCB1_v1.3_res_avg.tsv")



```


Data obtained from this: https://console.cloud.google.com/storage/browser/dm_alphamissense;tab=objects?pli=1&inv=1&invt=AbinwQ&prefix=&forceOnObjectsSortingFiltering=false
Downloaded this file: AlphaMissense_aa_substitutions.tsv.gz


```{r}
alpha_missense <- read_tsv(file=config$alphamissense_raw)


alpha_missense <- alpha_missense %>%
  mutate(
    Residue = as.numeric(str_extract(protein_variant, "\\d+")),  # Extract the number
    Amino_Acid_Alt = str_sub(protein_variant, start = -1)  # Extract the last letter
  )

alpha_missense <- alpha_missense %>%
  mutate(Residue = as.numeric(gsub("[^0-9]", "", protein_variant)))

alpha_missense$Mutation_type = "M"

write.table(alpha_missense, file = "./data/computational_predictors/SMARCB1_AlphaMissense_annotated.tsv", sep = "\t", row.names = FALSE, quote = FALSE)


pathogenic_alpha_missense <- alpha_missense %>%
  filter(alpha_missense$am_class == "pathogenic")

ambiguous_alpha_missense <- alpha_missense %>%
  filter(alpha_missense$am_class == "ambiguous")

benign_alpha_missense <- alpha_missense %>%
  filter(alpha_missense$am_class == "benign")

alpha_missense_average_residue <- alpha_missense %>%
  group_by(Residue) %>%
  summarise(res_avg_AM_score = mean(am_pathogenicity, na.rm = TRUE))

alpha_missense_average_residue_sorted <- alpha_missense_average_residue[order(-alpha_missense_average_residue$res_avg_AM_score), ]

alpha_missense_average_residue_sorted$rank <- 1:nrow(alpha_missense_average_residue_sorted)

write_tsv(alpha_missense_average_residue_sorted, "./data/computational_predictors/residue_averaged_zscores/AlphaMissense_SMARCB1_res_avg.tsv")

```

```{r}

library(purrr)

alpha_missense$AA_Substitution <- alpha_missense$protein_variant




merged_df <- full_join(alpha_missense, cadd_data_no_duplicates, by = "AA_Substitution")

merged_df2 <- full_join(merged_df, revel_data_no_duplicates, by = "AA_Substitution")

merged_df2 <- merged_df2 %>%
  mutate(Mutation_type_combined = pmap_chr(list(Mutation_type, Mutation_type.x, Mutation_type.y), function(x, y, z) {
    unique(c(x, y, z)) %>% # Combine and get unique values
      na.omit() %>%       # Remove NAs
      paste(collapse = "/") # Collapse into a single string
  })) %>%
  select(AA_Substitution, Mutation_type_combined, everything(), -Mutation_type, -Mutation_type.x, -Mutation_type.y)

merged_df3 <- merged_df2[,c(1,2,5,21,36)]

merged_df3 <- merged_df3 %>%
  rename(Mutation_type = Mutation_type_combined)

write_tsv(merged_df3, "./data/computational_predictors/computational_predictor_supp.tsv")

mite_data <- read_csv("./results/functional_scores/DMS_collapsed_data.csv")

mite_data <- mite_data %>%
  rename(AA_Substitution = Substitution)


merged_df4 <- left_join(merged_df3, mite_data, by = "AA_Substitution") %>%
  select(AA_Substitution, Mutation_type, am_pathogenicity, REVEL, PHRED, all_reps_zscore)


```


Plotting Correlation between CADD and DMS
```{r}

NMS_only <- merged_df4 %>%
  filter(merged_df4$Mutation_type == "M"| merged_df4$Mutation_type == "S"| merged_df4$Mutation_type == "Stop")

CADD_DMS_NMS <- na.omit(NMS_only[,c(1,2,5,6)])

write_tsv(CADD_DMS_NMS, "./results/functional_scores/CADD_DMS_correlation.tsv")

colors <- ifelse(CADD_DMS_NMS$Mutation_type == "S", "black",
                 ifelse(CADD_DMS_NMS$Mutation_type == "M", "gray", "red"))

tiff("./plots/computational_correlations/CADD_v_DMS_NMS.tiff", width = 10, height = 7, units = "in", res = 300)


plot(CADD_DMS_NMS$PHRED, CADD_DMS_NMS$all_reps_zscore,
     col = colors, pch = 19,  # pch = 19 for solid circles
     xlab = "PHRED", ylab = "All Reps Z-Score")

abline(h = 2, col = "black", lty = 2)  # Horizontal dashed line at Y = 2
abline(v = 20, col = "black", lty = 2)  # Vertical dashed line at X = 20

dev.off()


```

```{r}

N_only <- merged_df4 %>%
  filter(merged_df4$Mutation_type == "Stop")


N_pathogenic_DMS <-  N_only %>%
  filter(N_only$all_reps_zscore > 2)

N_pathogenic_CADD <-  N_only %>%
  filter(N_only$PHRED > 20)




S_only <- merged_df4 %>%
  filter(merged_df4$Mutation_type == "S")

S_pathogenic_DMS <-  S_only %>%
  filter(S_only$all_reps_zscore > 2)

S_pathogenic_CADD <-  S_only %>%
  filter(S_only$PHRED > 20)


M_only <- merged_df4 %>%
  filter(merged_df4$Mutation_type == "M")

M_only <- na.omit(M_only)

M_concordant <-  M_only %>%
  filter(M_only$all_reps_zscore > 2 & M_only$PHRED >20 | M_only$all_reps_zscore < 2 & M_only$PHRED < 20)


```



